manifests/peer.pp

```diff
@@ -35,13 +35,23 @@
 #   peering attempt only resolves a cosmetic issue, not a functional one.
 #
 define gluster::peer (
-  $pool = 'default',
-  $fqdn = $facts['networking']['fqdn'],
+  $pool                 = 'default',
+  $fqdn                 = $facts['networking']['fqdn'],
+  Boolean $force_binary = false,
 ) {
   # we can't do much without the Gluster binary
   # but we don't necessarily want the Puppet run to fail if the
   # gluster_binary fact is absent!
-  if getvar('::gluster_binary') {
+  if($force_binary) {
+    $real_binary = getvar('::gluster_binary') ? {
```

Comment:
what's the purpose of this workaround? The fact we have in https://github.com/voxpupuli/puppet-gluster/blob/master/lib/facter/gluster.rb checks if the binary gluster exists in $PATH.

Reply:
This getvar() stuff do not work at the first puppet run. So if the Server should be deployed with only 1 puppet agent run the glusterfs volume and peers won't be createt. With this workaround the binary variable will be set to the correct value so at the first run the peers an volumes could be created.

Reply:
while this approach might works, it's really tricky to read and I think we don't use a manual `lookup()` + hiera in modules just to get a single value. Instead of all this code we could add another parameter like this:
```puppet
Stdlib::Absolutepath $gluster_binary = pick($facts['gluster_binary'], '/usr/sbin/gluster')
```

Reply:
I will try the Stdlib version in my Test ENV.
I'll added the hiera stuff as a beginning to start removing the cases stuff from the pramas classes. But this i  would add in extra branches/PRs